### PR TITLE
cellSaveData_x_Delete method implementation

### PR DESCRIPTION
**Pull request by Dangles91**
Implemented cellSaveDataUserFixedDelete
Implemented cellSaveDataFixedDelete
Implemented msgDialog in cellSaveData for displaying confirmation messages and errors

Bug fix in msg_dialog_frame to set the correct background state of message dialogs (visible or transparent)

cellSaveDataListDelete is next on the the todo. Creating this PR to start broader testing and initial code review.

### CODE REVIEW

**Comment by Nekotekina** [comment]
Who needs confirmation dialogs?
Looks like a waste of time.
Both devs' and users'.

**Comment by Dangles91** [comment]
I see it as common courtesy. I don't imagine everyone will be a power user, and accidents do happen. 

Some games which use the fixed delete option will implement their own confirmation dialog, Minecraft for example, but games which rely on the system to provide that functionality will be without unless we implement it.

**Comment by Zardas81** [comment]
While accidents do happen, I'd say most of the time human actions are intended and only in small situations are they accidental. Immediate action with a chance to undo instead of ugly and annoying prompts would be better, like implementing some sort of 'marked for deletion' flag, and actual deletion happening when the dialog is closed or the program exits.

**Comment by spring568** [comment]
The dialogue boxes for saving and what not are annoying since one they don't work with a gamepad like the ps3 and two you can't see them when in fullscreen at least on windows. It'd be better do it as some kind of overlay instead

**Comment by Dangles91** [comment]
Marking something as deleted could work, simply renaming the save directory to prefix deleted_ would be enough. But I definitely think there needs to be some sort of fail safe, I'd hate to responsible for someone losing a save game they have put 10s of hours into when all they did was fat finger the keyboard or controller. 

@spring568 - definitely with you on the idea of an overlay, but unfortunately that's beyond my ability.

**Comment by fagoatse** [comment]
@Dangles91 I believe PPSSPP reimplements the save data dialog pretty much 1:1. Might be worth taking a look. iirc it's entirely GPU rendered.

**Comment by spring568** [comment]
Maybe, but even with code examples it would be difficult for someone who has little to no experience in that specific area of coding to implement it. He said himself that it was beyond his ability. Actually it was more of a general statement then being directed at Dangles91 so i feel a bit bad for just throwing it out like that

**Comment by Dangles91** [comment]
A note on the list delete confirmation dialog; in line with previous comments here I haven't included the final success dialog. I'm hoping this helps to illustrates the point of the confirmation dialogs, in that once you use the list delete you will likely find it somewhat jarring to not be notified that the delete was successful. 

When it comes to fixed delete, the application is able to specified whether or not it would like for the confirmation dialog to be displayed. This is configured via `fixedSet->option`.

So far I'm only aware of one game which uses the list delete feature; The Legend of Heroes Trails of Cold Steel.

File: rpcs3/rpcs3qt/msg_dialog_frame.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@ void msg_dialog_frame::Create(const std::string& msg)
 	m_dialog = new custom_dialog(type.disable_cancel);
 	m_dialog->setWindowTitle(type.se_normal ? "Normal dialog" : "Error dialog");
 	m_dialog->setWindowFlags(m_dialog->windowFlags() & ~Qt::WindowContextHelpButtonHint);
-	m_dialog->setWindowOpacity(type.bg_invisible ? 1.0 : 192.0 / 255.0);
+	m_dialog->setWindowOpacity(type.bg_invisible ? 192.0 / 255.0 : 1.0);
```
Comment (Nekotekina) [inline rpcs3/rpcs3qt/msg_dialog_frame.cpp]: What is this?

File: rpcs3/Emu/Cell/Modules/cellSaveData.h [hunk 2]
```diff
@@ -94,6 +93,13 @@ enum
 	CELL_SAVEDATA_RECREATE_MASK            = 0xffff,
 };
 
+// Dialog Messages
+namespace
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellSaveData.h]: This is horrible. Unnamed namespace in the header. Non-constexpr "constants". And all-caps style.

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp [hunk 3]
```diff
@@ -52,9 +56,55 @@ namespace
...
+			dlg_type = dlg_type | button_type;
+			dlg_type = dlg_type | msg_type;
+
+			MsgDialogType type = { dlg_type };
+
+			auto dlg = Emu.GetCallbacks().get_msg_dialog();
+			dlg->type = type;
+
+			// -2 is an unused enum value for CELL_MSGDIALOG_BUTTON_xx
+			s32 *ret_status = new s32(-2);
```
Comment (paulsapps) [inline rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: isn't this leaked?

File: rpcs3/rpcs3qt/msg_dialog_frame.cpp (hunk 1 above)
Comment (mchtt) [inline rpcs3/rpcs3qt/msg_dialog_frame.cpp]: looks like ip adresses to me lol

File: rpcs3/Emu/Cell/Modules/cellSaveData.h (hunk 2 above)
Comment (Dangles91) [inline rpcs3/Emu/Cell/Modules/cellSaveData.h]: Please let me know your preferred naming style. The rest can be fixed as well.

File: rpcs3/rpcs3qt/msg_dialog_frame.cpp (hunk 1 above)
Comment (Dangles91) [inline rpcs3/rpcs3qt/msg_dialog_frame.cpp]: This was mentioned in the initial commit message.
> Bug fix in msg_dialog_frame to set the correct background state of message dialogs (visible or transparent)

Bit 2 of the dialog type corresponds to the background style; a background style of CELL_MSGDIALOG_TYPE_BG_VISIBLE should display a solid background in the dialog window; 
CELL_MSGDIALOG_TYPE_BG_INVISIBLE should display no background (transparent).

File: rpcs3/Emu/Cell/Modules/cellSaveData.cpp (hunk 3 above)
Comment (Dangles91) [inline rpcs3/Emu/Cell/Modules/cellSaveData.cpp]: Indeed it is, I'll fix it up.

File: rpcs3/Emu/Cell/Modules/cellSaveData.h (hunk 2 above)
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/cellSaveData.h]:
```
constexpr auto s_abc = "abc";
```

**Comment by Nekotekina** [comment]
Closed as heavily outdated. Core save delete functionality is now implemented in master. Feel free to reopen or create another PR if it something gets reworked.

**Pull request closed without merging**